### PR TITLE
Added the Geolocation (current location of user) Feature

### LINKED ISSUE #4: Add a 'Current Location' button to the map
## User story

As a transit rider,
I want to easily see my current location on the map,
So that I can find nearby stops.

## Requirements

- [x] add a 'Find Current Location' button to the map using the [location-crosshairs](https://fontawesome.com/icons/location-crosshairs?f=classic&s=solid) icon. See StopPane.svelte for an example of how to use these. 
- [x] Clicking the button should prompt the user for permission to their location and, if granted, re-center the map on the user's location and show a dot on the map indicating the user's location.

## UI

The button should be located roughly where the purple box indicates.

Screenshot 2024-06-22 at 9 05 44 AM

### PR DESCRIPTION
**Pull request by tarunsinghofficial**
Fixed #4 

**Tasks done**:

- Added a 'Find Current Location' button to the map.
- Clicking the button first gets the user permission, and if allowed we center the map with their current location.

**Screenshot:**
bandicam2024-07-0520-55-03-272-ezgif com-video-to-gif-converter

### CODE REVIEW

File: src/components/map/GoogleMap.svelte [hunk 1]
```diff
@@ -104,6 +142,10 @@
 	});
 </script>
 
+<link
```
Comment (aaronbrethorst) [inline src/components/map/GoogleMap.svelte:145]: make sure to remove this after making the other changes!

**Review by aaronbrethorst — CHANGES_REQUESTED** [review]
The feature works great! I noted some code changes I'd like to see before merging. Please let me know via slack if you have any questions!

File: src/components/map/GoogleMap.svelte [hunk 2]
```diff
@@ -44,6 +45,43 @@
...
 		}
+
+		const locationButton = document.createElement('button');
+
+		locationButton.innerHTML = '<i class="fa-solid fa-location-crosshairs"></i>';
+		locationButton.classList.add('custom-map-control-button');
+		document.getElementById('map').appendChild(locationButton);
+
+		locationButton.addEventListener('click', () => {
+			if (navigator.geolocation) {
```
[the remark this answers is not in the document]
Comment (tarunsinghofficial) [reply src/components/map/GoogleMap.svelte:56]: Sure, that's a great approach. I have added the early return statement.

File: src/components/map/GoogleMap.svelte [hunk 3]
```diff
@@ -44,6 +45,43 @@
 		if (browser) {
 			window.addEventListener('themeChange', handleThemeChange);
 		}
+
+		const locationButton = document.createElement('button');
```
[the remark this answers is not in the document]
Comment (tarunsinghofficial) [reply src/components/map/GoogleMap.svelte:49]: Okay. I will make sure to create a reusable component for using in the future.

File: src/components/map/GoogleMap.svelte [hunk 4]
```diff
@@ -44,6 +45,43 @@
...
+
+		locationButton.addEventListener('click', () => {
+			if (navigator.geolocation) {
+				navigator.geolocation.getCurrentPosition(
+					(position) => {
+						const { latitude, longitude } = position.coords;
+						const userLocation = new google.maps.LatLng(latitude, longitude);
+						map.setCenter(userLocation);
+
+						new google.maps.Marker({
```
[the remark this answers is not in the document]
Comment (tarunsinghofficial) [reply src/components/map/GoogleMap.svelte:63]: Done this one.

**Review by aaronbrethorst — APPROVED** [review]
awesome, great work! 👏

**Pull request merged**